clang/include/clang/Sema/Sema.h

```diff
@@ -10094,8 +10094,8 @@ class Sema final {
   void AddIntelFPGABankBitsAttr(Decl *D, const AttributeCommonInfo &CI,
                                 Expr **Exprs, unsigned Size);
   template <typename AttrType>
-  void addIntelSYCLSingleArgFunctionAttr(Decl *D, const AttributeCommonInfo &CI,
-                                         Expr *E);
+  void addSingleArgFunctionAttr(Decl *D, const AttributeCommonInfo &CI,
```

Comment:
I think it's actually a bit dangerous to rename this function -- it's not a generic function that can be used with any attribute because the function body itself is looking at the contents of specific parsed attribute kinds to decide what to do. So I'd say this function name should be left alone.

My hope is to eventually refactor this function either out of existence, or to make it a `static` function that's internal to SemaDeclAttr.cpp.

Reply:
I removed the 'SYCL' and 'Function' from the name since I now use this function to handle other FPGA attributes as well. I'm not a fan of the new name `addIntelSingleArgAttr` but I'm not sure what else to call it. Please feel free to suggest another name or let me know if you'd like me to refactor this function in some way. I think the diagnostics should probably be moved out to the individual attribute handlers, but I think this work overlaps with the patch you're working on and so I left it as is.  

I think another look into this function, `AddOneConstantValueAttr `, `AddOneConstantPowerTwo `, `instantiateIntelFPGAMemoryAttr `and `instantiateIntelSYCLFunctionAttr` is required  at some point. All of these could possibly be refactored, maybe even removed.

Reply:
My long-term goal is to remove these functions that look like they should be generic but actually aren't generic. Ideally, we could use tablegen to generate code that turns out to be boilerplate in practice. So the name isn't super worrying right now by comparison to the function existing. :-D